components/templates/CodingLayout.tsx

```diff
@@ -0,0 +1,86 @@
+'use client'
+
+import { allLessons, Lesson } from 'contentlayer/generated'
+import { useState, useEffect } from 'react'
+import { useMediaQuery } from 'react-responsive'
+import { Tabs } from 'components/ui/Tabs'
+import { Info } from './Info'
+import { Code } from './Code'
+
+/**
+ * @slug {string} for fetching challenge data
+ * @solutionScriptSigHex {string} answer to the challenge problem
+ * @next {string} link to next part of chapter
+ */
+export default function CodingLayout({ 
+        slug, 
```

Comment:
For progress tracking, the page also needs to know which chapter it's in.

We also want an extra parameter to customize the success message.

And potentially another one for the tips feature, see #56.

Reply:
Based on your suggestions we can have the following parameters:
- `chapterName` : full chapter name
- `successMessage` : success message
- `tips` : this can be an array in case we have multiple tips for a challenge